### PR TITLE
130 bootstrap documentation

### LINKED ISSUE #130: Documentation: choose documentation framework, bootstrap documentation

### PR DESCRIPTION
**Pull request by gboutry**
fix #130 

this pr:
- uses Sphinx as a documentation framework
- sets some ground work, few pages with minimal explanation, more work shall be done in #97
- Some examples on how to document code, more work shall be done in #131

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -21,19 +21,32 @@ GitPython = "3.1.18"
 SQLAlchemy = "^1.4.31"
 click = "8.0.4"
 tabulate = "0.8.9"
+Sphinx = {version = "4.5.0", optional = true}
+sphinx-markdown-builder = {version = "0.5.5", optional = true}
```
Comment (rpignolet) [inline pyproject.toml:25]: This dependency is used ?

Comment (gboutry) [reply pyproject.toml]: It's used in markdown generation command, I removed it in the latest commit

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
+.. TDP documentation master file, created by
```
[the remark this answers is not in the document]
Comment (gboutry) [reply docs/index.rst]: Done

**Comment by gboutry** [comment]
rebase on master

**Comment by leopaul36** [comment]
I pushed some docs generation instructions

**Review by leopaul36 — APPROVED** [review]

**Pull request merged**